atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java

```diff
@@ -78,14 +78,15 @@ public CassandraClientPoolingContainer(
             CassandraServer cassandraServer,
             CassandraKeyValueServiceConfig config,
             int poolNumber,
-            CassandraClientPoolMetrics poolMetrics) {
+            CassandraClientPoolMetrics poolMetrics,
+            CassandraClientInstrumentation cassandraClientInstrumentation) {
         this.metricsManager = metricsManager;
         this.cassandraServer = cassandraServer;
         this.proxy = cassandraServer.proxy();
         this.config = config;
         this.poolNumber = poolNumber;
         this.poolMetrics = poolMetrics;
-        this.clientPool = createClientPool();
+        this.clientPool = createClientPool(cassandraClientInstrumentation);
```

Comment:
This class's setup is a bit annoying. I spent some time suffering with a bug I had here where `CassandraClientInstrumentation` was a field and I had written things this way
```
this.clientPool = createClientPool();
this.cassandraClientPoolInstrumentation = cassandraClientPoolInstrumentation;
```

And, `createClientPool` used the instrumentation object. You can see how this can result in issues.

Ideally, `createClientPool` should be a static. I tried, but it looked like a larger refactor with many static methods, enough so that I think the nice way to do it is to extract the static logic to its own class.

The PR adds the new dependency in a non-error prone way and I think this is enough.